rfcs/text/0015_nodejs_clustering.md

```diff
@@ -0,0 +1,554 @@
+  for the coordinator process
+- Even if we add the `%worker` pattern, do we want to also allow users to specify per-worker log files? 
+
+### The rolling-file appender
+
+The rolling process of the `rolling-file` appender is going to be problematic in clustered mode, as it will cause 
+concurrency issues during the rolling. We need to find a way to have this rolling stage clustered-proof.
+
+Identified options are:
+
+1. have the rolling file appenders coordinate themselves when rolling
+
+By using a broadcast message based mutex mechanism, the appenders could acquire a ‘lock’ to roll a specific file, and
+notify other workers when the rolling is complete (quite similar to what we want to do with SO migration for example).
+
+An alternative to this option would be to only have the main worker handle the rolling logic. We will lose control
+on the exact size the file is when rolling, as we would need to wait until the main worker receives a log message
+for the rolling appender before the rolling is effectively performed. The upside would be that it reduces the inter-workers
+communication to a notification from the main worker to the others once the rolling is done for them to reopen their
+file handler.
+
+2. have the coordinator process perform the rolling
+
+Another option would be to have the coordinator perform the rotation instead. When a rolling is required, the appender 
+would send a message to the coordinator, which would perform the rolling and notify the workers once the operation is complete. 
+
+Note that this option is even more complicated than the previous one, as it forces to move the rolling implementation 
+outside of the appender, without any significant upsides identified.
+
+3. centralize the logging system in the coordinator
+
+We could go further, and change the way the logging system works in clustering mode by having the coordinator centralize 
+the logging system. The worker’s logger implementation would just send messages to the coordinator. If this may be a 
+correct design, the main downside is that the logging implementation would be totally different in cluster and 
+non cluster mode, and seems to be way more work that the other options.
+
+Overall, if no option is trivial, I feel like option `1.` is still the most pragmatic one. Option `3.` may be a better 
+design, but represents more work. It would probably be fine if all our appenders were impacted, but as only the 
+rolling-file one is, I feel like going with `3.` would be ok.
+
+### The status API
+
+In clustering mode, the workers will all have an individual status. One could have a connectivity issue with ES 
+while the other ones are green. Hitting the `/status` endpoint will reach a random (and different each time) worker, 
+meaning that it would not be possible to know the status of the cluster as a whole.
+
+We will need to add a centralized status service in the coordinator. Also, as the `/status` endpoint cannot be served 
```

Comment:
Will the coordinator serve any HTTP requests, or will they all be served by the workers themselves?